src/snowflake/snowpark/_internal/code_generation.py

```diff
@@ -180,12 +180,12 @@ def get_class_references(
 
 def extract_func_global_refs(code: CodeType) -> Set[str]:
     # inspired by cloudpickle to recursively extract all the global references used by the target func's code object
-    co_names = code.co_names
+    co_names = set(code.co_names)
     out_names = set()
     for instr in dis.get_instructions(code):
         op = instr.opcode
-        if op in GLOBAL_OPS:
-            out_names.add(co_names[instr.arg])
+        if op in GLOBAL_OPS and instr.argval in co_names:
+            out_names.add(instr.argval)
```

Comment:
Sanity check here: Is it the case that previously `co_names` is a tuple/list and `instr.arg` is an index, and now we changed it such that we directly add value if it exists?

Is there a case where `instr.argval in co_names` can be `False`? This seems to be a logic change given previously we just check if `op` is in `GLOBAL_OPS`.

Reply:
thanks for the sanity check -- I was copying the change from the external PR: https://github.com/snowflakedb/snowpark-python/pull/857 but I think you are right.

after double checking cloudpickle's implementation, the co_names is not used anymore and only `argval` is used: https://github.com/cloudpipe/cloudpickle/commit/6a0e12d058d1bd3ab26ec000ac2249b4ee7e9c9f.

I have updated the code to align with cloudpickle's latest implementation.